records_mover/records/pandas/format_for_csv.py

```diff
@@ -0,0 +1,62 @@
+from pandas import DataFrame
+import pandas as pd
+from records_mover.records.schema import RecordsSchema
+from records_mover.records import DelimitedRecordsFormat
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def format_df_for_csv_output(df: DataFrame,
+                             records_schema: RecordsSchema,
+                             records_format: DelimitedRecordsFormat) -> DataFrame:
+    formatted_df = df.copy(deep=False)
+
+    logger.info(f"VMB formatted_df: {formatted_df}")
+    logger.info(f"VMB index, field: {list(enumerate(records_schema.fields))}")
+    logger.info(f"VMB formatted_df.iloc[:, 1]: {formatted_df.iloc[:, 1]}")
+    for index, field in enumerate(records_schema.fields):
+        logger.info(f"VMB: Looking at field {field.name} with type {field.field_type}")
+        if field.field_type == 'date':
+            # TODO: is there a less weird way of doing this?
+            series = formatted_df.iloc[:, index]
+            if not isinstance(series[0], pd.Timestamp):
+                logger.warning(f"Found {series.name} as unexpected type {type(series[0])}")
+            else:
+                logger.info(f"Converting {series.name} from np.datetime64 to string in CSV's format")
+                hint_date_format = records_format.hints['dateformat']
+                assert isinstance(hint_date_format, str)
+                # TODO: Do I have this conversion elsewhere?
+                pandas_date_format_conversion = {
+                    'YYYY-MM-DD': '%Y-%m-%d',
+                    'MM/DD/YY': '%m/%d/%Y',
+                    'DD/MM/YY': '%d/%m/%Y',
+                }
+                pandas_date_format = pandas_date_format_conversion.get(hint_date_format,
+                                                                       '%Y-%m-%d')
+                formatted_series = series.dt.strftime(pandas_date_format)
+                formatted_df.iloc[:, index] = formatted_series
+                # logger.info(f"VMB formatted_series: {formatted_series}")
+                # logger.info(f"VMB formatted_df['date']: {formatted_df['date']}")
+                # logger.info(f"VMB formatted_df: {formatted_df}")
+        elif field.field_type == 'time':
+            # TODO: What about timetz
+            # TODO: is there a less weird way of doing this?
+            series = formatted_df.iloc[:, index]
+            if not isinstance(series[0], pd.Timestamp):
+                logger.warning(f"Found {series.name} as unexpected type {type(series[0])}")
+            else:
+                logger.info(f"Converting {series.name} from np.datetime64 to string in CSV's format")
+                hint_time_format = records_format.hints['timeonlyformat']
+                assert isinstance(hint_date_format, str)
+                # TODO: Do I have this conversion elsewhere?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker